src/ansys/simai/core/client.py

```diff
@@ -68,7 +69,14 @@ class SimAIClient:
 
     @steal_kwargs_type(ClientConfig)
     def __init__(self, **kwargs):
-        config = ClientConfig(**kwargs)
+        try:
+            config = ClientConfig(**kwargs)
+        except ValidationError as pyandic_exc:
+            missing_properties = [exc["msg"] for exc in pyandic_exc.errors()]
```

Comment:
Can we check for which kind of error it is ? Some errors might not be missing properties.

Looking at some code I found in the pydantic doc (https://docs.pydantic.dev/latest/api/pydantic_extra_types_mac_address/#pydantic_extra_types.mac_address.MacAddress.validate_mac_address , the source code at the very bottom), I think it would be best to have an error message for each error we generate and let pydantic generate the full error message.
That being said catching that and reraising an `InvalidConfigurationError` like you did is the right thing to do.

Reply:
implementation is updated